content/ember-data-polaris-2023-04-11.md

```diff
@@ -0,0 +1,73 @@
+---
+title: EmberData Polaris Update
+authors:
+  - chris-thoburn
+date: 2023-04-11T00:00:00.000Z
+tags:
+  - '2023'
+  - ember-data
+  - ember-polaris
+
+---
+
+This week, in coordination with the broader Ember project, EmberData released 4.12, its final
+4.x version, and began iterating towards the first release of the 5.x series.
+
+## A wrap on 4.x
+
+This final release of the 4.x cycle introduced several new concepts to support a new paradigm for managing [Requests]() and new capabilities for [Caching Data]().
+
+The primary focus of these changes was to support the concept of a [Document]() as a cache primitive and to complete the work to ensure that the [Resource]() concept is handled opaquely.
+
+Collectively, these changes allow applications the capability of building highly advanced request and cache capabilities overtop of familiar EmberData paradigms. We suggest reading the respective RFCs for [RequestManager]() and [Cache]() to gain a fuller appreciation of the motivations and capabilities they bring.
+
+While these new primitives may not at first feel like a substantial shift, over the course of 5.x as the new request story is polished we expect for the recommended experience when using EmberData to significantly evolve, with that evolution driven by these changes.
+
+## Looking ahead to 5.x
+
+Throughout the 5.x cycle we expect to introduce two important new defaults while deprecating an old stalwart.
+
+For nearly 17 years, [Model](https://github.com/sproutcore/sproutcore/commit/f6248b1650a688a401cc6eea135fbe983e20cd12#diff-011979c89114a908391f35c2053dc2ba84da4d331cc97730039b2b2da623ffee) has been a foundational primitive around which EmberData was understood. Since those earliest days, the language and ecosystem have evolved, the kinds of applications we build has evolved, and the patterns by which we access and mutate data have evolved. While Model has undergone small amounts of evolution in syntax, its underlying patterns have remained unevolved.
+
+In more recent years, various 3rd party attempts have been made to address some of the shortcomings of the Model paradigm. [ember-m3]() explored what it might mean to have models be schema-driven. [ember-data-model-fragments]() offered a workaround for deep-tracking. [ember-data-storefront]() offered alternative data access patterns to simplify the mental model of working with asyncronous edges in relational data. [ember-data-changetracker](), [ember-changesets](), and [ember-buffered-proxy]() offered mechanisms for attempting to streamline mutation flows and make changes more easily discardable.
```

Comment:
```suggestion
In more recent years, various 3rd party attempts have been made to address some of the shortcomings of the Model paradigm. [ember-m3]() explored what it might mean to have schema-driven models. [ember-data-model-fragments]() offered a deep-tracking workaround. [ember-data-storefront]() offered alternative data access patterns to simplify the mental model of asyncronous edges in relational data. [ember-data-changetracker](), [ember-changesets](), and [ember-buffered-proxy]() offered mechanisms for streamlined mutation flows and more easily discardable changes.
```